tests/unit/meta/test_snap_yaml.py

```diff
@@ -502,3 +503,24 @@ def test_project_environment_ld_library_path_null(simple_project, new_dir):
           PATH: $SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH
         """
     )
+
+
+def test_version_git(simple_project, new_dir, mocker):
```

Comment:
this might be better in spread where the git setup is somewhat already handled, for units just ensure the externalities are  called (GitSource)

Reply:
Mocking `generate_version`, craft-parts tests should ensure it's working as expected.

Reply:
Added a spread test.